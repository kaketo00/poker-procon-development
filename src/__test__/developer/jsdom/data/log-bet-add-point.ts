import type { Log } from '@/schema/log';

export const testLogBetAddPoint: Log = {
  round: 10,
  player: 'Player1',
  phase: 'bet-1',
  content: { type: 'bet', amount: 314, action: 'bet', payPoint: 314 },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 64746,
        round: {
          betPoint: 314,
          first: 314,
          second: 0,
          action: 'bet',
          cards: [
            { suit: 'Clubs', number: 6 },
            { suit: 'Spades', number: 8 },
            { suit: 'Hearts', number: 9 },
            { suit: 'Diamonds', number: 1 },
            { suit: 'Spades', number: 1 },
          ],
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 6256,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 6 },
            { suit: 'Clubs', number: 7 },
            { suit: 'Diamonds', number: 7 },
            { suit: 'Diamonds', number: 12 },
            { suit: 'Clubs', number: 1 },
          ],
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, first: 0, second: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 8084,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 3 },
            { suit: 'Hearts', number: 4 },
            { suit: 'Diamonds', number: 4 },
            { suit: 'Spades', number: 11 },
            { suit: 'Hearts', number: 11 },
          ],
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 914,
  minBetPoint: 314,
  createdAt: 1713787617161,
};
