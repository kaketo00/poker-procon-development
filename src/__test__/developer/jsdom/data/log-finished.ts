import type { Log } from '@/schema/log';

export const testLogFinished: Log = {
  round: 10,
  player: '',
  phase: 'finished',
  content: {
    type: 'finished',
    pot: 7461,
    winner: 'Player1',
    result: {
      Player1: {
        cards: [
          { suit: 'Spades', number: 4 },
          { suit: 'Clubs', number: 4 },
          { suit: 'Clubs', number: 5 },
          { suit: 'Diamonds', number: 8 },
          { suit: 'Diamonds', number: 13 },
        ],
        hand: 2,
      },
      Player4: {
        cards: [
          { suit: 'Hearts', number: 2 },
          { suit: 'Clubs', number: 3 },
          { suit: 'Hearts', number: 8 },
          { suit: 'Clubs', number: 10 },
          { suit: 'Spades', number: 11 },
        ],
        hand: 1,
      },
      Player3: { cards: [], hand: 0 },
      Player2: {
        cards: [
          { suit: 'Clubs', number: 2 },
          { suit: 'Spades', number: 3 },
          { suit: 'Diamonds', number: 5 },
          { suit: 'Diamonds', number: 10 },
          { suit: 'Clubs', number: 12 },
        ],
        hand: 1,
      },
    },
  },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 70234,
        round: {
          betPoint: 2287,
          action: 'call',
          cards: [
            { suit: 'Spades', number: 4 },
            { suit: 'Clubs', number: 4 },
            { suit: 'Clubs', number: 5 },
            { suit: 'Diamonds', number: 8 },
            { suit: 'Diamonds', number: 13 },
          ],
          hand: 2,
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 3969,
        round: {
          betPoint: 2287,
          action: 'call',
          cards: [
            { suit: 'Clubs', number: 2 },
            { suit: 'Spades', number: 3 },
            { suit: 'Diamonds', number: 5 },
            { suit: 'Diamonds', number: 10 },
            { suit: 'Clubs', number: 12 },
          ],
          hand: 1,
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 5797,
        round: {
          betPoint: 2287,
          action: 'raise',
          cards: [
            { suit: 'Hearts', number: 2 },
            { suit: 'Clubs', number: 3 },
            { suit: 'Hearts', number: 8 },
            { suit: 'Clubs', number: 10 },
            { suit: 'Spades', number: 11 },
          ],
          hand: 1,
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 7461,
  minBetPoint: 2287,
  createdAt: 1713787617170,
};
