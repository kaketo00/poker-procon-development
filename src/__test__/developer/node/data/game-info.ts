import type { GameInfo } from '@/schema/game';

export const testGameInfoData: GameInfo = {
  currentRound: 1,
  phase: 'start',
  order: ['Player1', 'Player2', 'Player3', 'Player4'],
  pot: 0,
  minBetPoint: 0,
  players: {
    Player1: {
      name: 'Player1',
      status: 'active',
      point: 37600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player2: {
      name: 'Player2',
      status: 'active',
      point: 17600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player3: {
      name: 'Player3',
      status: 'active',
      point: 24800,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player4: {
      name: 'Player4',
      status: 'out',
      point: 0,
      round: {
        betPoint: 0,
        action: null,
        cards: [],
      },
    },
  },
  totalRound: 100,
  initialPoint: 20000,
  fee: 200,
};
