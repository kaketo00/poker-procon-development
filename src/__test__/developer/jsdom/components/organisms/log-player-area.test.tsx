import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';

import { testLogBetAddPoint } from '@/__test__/developer/jsdom/data/log-bet-add-point';
import { testLogDraw } from '@/__test__/developer/jsdom/data/log-draw';
import { testLogFinished } from '@/__test__/developer/jsdom/data/log-finished';
import type { LogPlayerAreaProps } from '@/components/organisms/log-player-area';
import LogPlayerArea from '@/components/organisms/log-player-area';
import theme from '@/styles/theme';

const Component = (props: LogPlayerAreaProps) => (
  <ChakraProvider theme={theme}>
    <LogPlayerArea {...props} />
  </ChakraProvider>
);

describe('LogPlayerArea', () => {
  test('スナップショット/bet phase', () => {
    const { asFragment } = render(
      <Component
        player={{
          name: 'Player1',
          status: 'active',
          point: 19358,
          round: {
            betPoint: 442,
            first: 442,
            second: 0,
            cards: [
              { suit: 'Spades', number: 4 },
              { suit: 'Clubs', number: 9 },
              { suit: 'Spades', number: 10 },
              { suit: 'Spades', number: 11 },
              { suit: 'Hearts', number: 12 },
            ],
            action: 'bet',
          },
        }}
        log={testLogBetAddPoint}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/draw phase', () => {
    const { asFragment } = render(
      <Component
        player={{
          name: 'Player1',
          status: 'active',
          point: 19358,
          round: {
            betPoint: 442,
            first: 442,
            second: 0,
            cards: [
              { suit: 'Spades', number: 4 },
              { suit: 'Clubs', number: 9 },
              { suit: 'Spades', number: 10 },
              { suit: 'Spades', number: 11 },
              { suit: 'Hearts', number: 12 },
            ],
            action: 'bet',
          },
        }}
        log={testLogDraw}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/自分のターンではない場合', () => {
    const { asFragment } = render(
      <Component
        player={{
          name: 'Player2',
          status: 'active',
          point: 19358,
          round: {
            betPoint: 442,
            first: 442,
            second: 0,
            cards: [
              { suit: 'Spades', number: 4 },
              { suit: 'Clubs', number: 9 },
              { suit: 'Spades', number: 10 },
              { suit: 'Spades', number: 11 },
              { suit: 'Hearts', number: 12 },
            ],
            action: 'bet',
          },
        }}
        log={testLogDraw}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/out', () => {
    const { asFragment } = render(
      <Component
        player={{
          name: 'Player3',
          status: 'out',
          point: 0,
          round: {
            betPoint: 0,
            first: 0,
            second: 0,
            cards: [],
            action: null,
          },
        }}
        log={testLogDraw}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/finished phase - active', () => {
    const { asFragment } = render(
      <Component
        player={{
          name: 'Player1',
          status: 'active',
          point: 19358,
          round: {
            betPoint: 442,
            first: 442,
            second: 0,
            cards: [
              { suit: 'Spades', number: 4 },
              { suit: 'Clubs', number: 9 },
              { suit: 'Spades', number: 10 },
              { suit: 'Spades', number: 11 },
              { suit: 'Hearts', number: 12 },
            ],
            action: 'call',
            hand: 1,
          },
        }}
        log={testLogFinished}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/finished phase - out', () => {
    const { asFragment } = render(
      <Component
        player={{
          name: 'Player2',
          status: 'out',
          point: 0,
          round: {
            betPoint: 0,
            first: 0,
            second: 0,
            cards: [],
            action: null,
          },
        }}
        log={testLogFinished}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
