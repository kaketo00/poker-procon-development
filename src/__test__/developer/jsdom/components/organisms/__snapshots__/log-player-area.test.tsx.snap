// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogPlayerArea スナップショット/bet phase 1`] = `
<DocumentFragment>
  <div
    class="css-dvxtzn"
  >
    <div
      class="css-79elbk"
    >
      <div
        class="css-aryzif"
      >
        <div
          class="css-shjrb7"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-04.svg"
          />
        </div>
        <div
          class="css-fni08i"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Clubs-09.svg"
          />
        </div>
        <div
          class="css-1bd1kgk"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-10.svg"
          />
        </div>
        <div
          class="css-2ikapx"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-11.svg"
          />
        </div>
        <div
          class="css-16c4tub"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Hearts-12.svg"
          />
        </div>
      </div>
      <div
        class="css-8myd60"
      >
        <div
          class="css-19tvp9s"
        >
          <p
            class="chakra-text css-1876eo0"
          >
            High Card
          </p>
          <img
            alt="bubble"
            class="chakra-image css-13pr30p"
            src="/bubble.svg"
          />
        </div>
        <div
          class="css-70qvj9"
        >
          <div
            class="css-1mt82r2"
          >
            <div
              class="css-o3ijz8"
            >
              <svg
                fill="none"
                height="72"
                viewBox="0 0 61 72"
                width="61"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.4336 17.614C12.4336 7.88457 20.4352 0 30.3091 0C40.1829 0 48.1905 7.88457 48.1905 17.614C48.1905 27.3492 40.1829 35.2338 30.3091 35.2338C20.4352 35.2338 12.4336 27.3492 12.4336 17.614ZM30.4429 72.0003C42.2478 72.0003 52.931 67.3433 60.6312 59.8513V54.5366C60.6312 48.1099 55.2047 42.8941 48.505 42.8941H12.1262C5.42649 42.8941 0 48.1099 0 54.5366V59.5836C7.71835 67.2269 18.5046 72.0003 30.4429 72.0003Z"
                  fill="#005CFF"
                  fill-opacity="0.8"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-8pxmcc"
          >
            <p
              class="chakra-text css-1j4e3xu"
            >
              Player1
            </p>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-1qn2z3z"
            />
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1qwa78t"
              >
                19,358
              </p>
              <p
                class="chakra-text css-84i6iv"
              >
                pt
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-7xthq6"
    >
      <p
        class="chakra-text css-w1ydsw"
      >
        Bet +314pt
      </p>
    </div>
    <div
      class="css-z4tay6"
    >
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          fee
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          200
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          1st bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          442
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          2nd bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          0
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        =
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          total
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          642
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        pt
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`LogPlayerArea スナップショット/draw phase 1`] = `
<DocumentFragment>
  <div
    class="css-dvxtzn"
  >
    <div
      class="css-79elbk"
    >
      <div
        class="css-aryzif"
      >
        <div
          class="css-shjrb7"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-04.svg"
          />
        </div>
        <div
          class="css-fni08i"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Clubs-09.svg"
          />
        </div>
        <div
          class="css-1bd1kgk"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-10.svg"
          />
        </div>
        <div
          class="css-2ikapx"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-11.svg"
          />
        </div>
        <div
          class="css-16c4tub"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Hearts-12.svg"
          />
        </div>
      </div>
      <div
        class="css-8myd60"
      >
        <div
          class="css-19tvp9s"
        >
          <p
            class="chakra-text css-1876eo0"
          >
            One Pair → Three of a Kind
          </p>
          <img
            alt="bubble"
            class="chakra-image css-13pr30p"
            src="/bubble.svg"
          />
        </div>
        <div
          class="css-70qvj9"
        >
          <div
            class="css-1mt82r2"
          >
            <div
              class="css-o3ijz8"
            >
              <svg
                fill="none"
                height="72"
                viewBox="0 0 61 72"
                width="61"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.4336 17.614C12.4336 7.88457 20.4352 0 30.3091 0C40.1829 0 48.1905 7.88457 48.1905 17.614C48.1905 27.3492 40.1829 35.2338 30.3091 35.2338C20.4352 35.2338 12.4336 27.3492 12.4336 17.614ZM30.4429 72.0003C42.2478 72.0003 52.931 67.3433 60.6312 59.8513V54.5366C60.6312 48.1099 55.2047 42.8941 48.505 42.8941H12.1262C5.42649 42.8941 0 48.1099 0 54.5366V59.5836C7.71835 67.2269 18.5046 72.0003 30.4429 72.0003Z"
                  fill="#005CFF"
                  fill-opacity="0.8"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-8pxmcc"
          >
            <p
              class="chakra-text css-1j4e3xu"
            >
              Player1
            </p>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-1qn2z3z"
            />
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1qwa78t"
              >
                19,358
              </p>
              <p
                class="chakra-text css-84i6iv"
              >
                pt
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-7xthq6"
    >
      <p
        class="chakra-text css-8eqocq"
      >
        Exchange 1 cards
      </p>
    </div>
    <div
      class="css-z4tay6"
    >
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          fee
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          200
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          1st bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          442
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          2nd bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          0
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        =
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          total
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          642
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        pt
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`LogPlayerArea スナップショット/finished phase - active 1`] = `
<DocumentFragment>
  <div
    class="css-dvxtzn"
  >
    <img
      alt="win"
      class="chakra-image css-tz7uiz"
      src="/win.svg"
    />
    <div
      class="css-79elbk"
    >
      <div
        class="css-aryzif"
      >
        <div
          class="css-shjrb7"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-04.svg"
          />
        </div>
        <div
          class="css-fni08i"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Clubs-09.svg"
          />
        </div>
        <div
          class="css-1bd1kgk"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-10.svg"
          />
        </div>
        <div
          class="css-2ikapx"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-11.svg"
          />
        </div>
        <div
          class="css-16c4tub"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Hearts-12.svg"
          />
        </div>
      </div>
      <div
        class="css-8myd60"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="css-1mt82r2"
          >
            <div
              class="css-o3ijz8"
            >
              <svg
                fill="none"
                height="72"
                viewBox="0 0 61 72"
                width="61"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.4336 17.614C12.4336 7.88457 20.4352 0 30.3091 0C40.1829 0 48.1905 7.88457 48.1905 17.614C48.1905 27.3492 40.1829 35.2338 30.3091 35.2338C20.4352 35.2338 12.4336 27.3492 12.4336 17.614ZM30.4429 72.0003C42.2478 72.0003 52.931 67.3433 60.6312 59.8513V54.5366C60.6312 48.1099 55.2047 42.8941 48.505 42.8941H12.1262C5.42649 42.8941 0 48.1099 0 54.5366V59.5836C7.71835 67.2269 18.5046 72.0003 30.4429 72.0003Z"
                  fill="#005CFF"
                  fill-opacity="0.8"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-8pxmcc"
          >
            <p
              class="chakra-text css-1j4e3xu"
            >
              Player1
            </p>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-1qn2z3z"
            />
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1qwa78t"
              >
                19,358
              </p>
              <p
                class="chakra-text css-84i6iv"
              >
                pt
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-7xthq6"
    >
      <p
        class="chakra-text css-8eqocq"
      >
        High Card
      </p>
    </div>
    <div
      class="css-z4tay6"
    >
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          fee
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          200
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          1st bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          442
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          2nd bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          0
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        =
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          total
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          642
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        pt
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`LogPlayerArea スナップショット/finished phase - out 1`] = `
<DocumentFragment>
  <div
    class="css-dvxtzn"
  >
    <div
      class="css-79elbk"
    >
      <div
        class="css-aryzif"
      />
      <div
        class="css-8myd60"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="css-e18jq9"
          >
            <div
              class="css-o3ijz8"
            >
              <svg
                fill="none"
                height="72"
                viewBox="0 0 61 72"
                width="61"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.4336 17.614C12.4336 7.88457 20.4352 0 30.3091 0C40.1829 0 48.1905 7.88457 48.1905 17.614C48.1905 27.3492 40.1829 35.2338 30.3091 35.2338C20.4352 35.2338 12.4336 27.3492 12.4336 17.614ZM30.4429 72.0003C42.2478 72.0003 52.931 67.3433 60.6312 59.8513V54.5366C60.6312 48.1099 55.2047 42.8941 48.505 42.8941H12.1262C5.42649 42.8941 0 48.1099 0 54.5366V59.5836C7.71835 67.2269 18.5046 72.0003 30.4429 72.0003Z"
                  fill="#005CFF"
                  fill-opacity="0.8"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-1ch8wrx"
          >
            <p
              class="chakra-text css-1j4e3xu"
            >
              Player2
            </p>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-1qn2z3z"
            />
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1qwa78t"
              >
                0
              </p>
              <p
                class="chakra-text css-84i6iv"
              >
                pt
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-z4tay6"
    >
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          fee
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          200
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          1st bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          0
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          2nd bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          0
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        =
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          total
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          200
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        pt
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`LogPlayerArea スナップショット/out 1`] = `
<DocumentFragment>
  <div
    class="css-dvxtzn"
  >
    <div
      class="css-79elbk"
    >
      <div
        class="css-aryzif"
      />
      <div
        class="css-8myd60"
      >
        <div
          class="css-70qvj9"
        >
          <div
            class="css-e18jq9"
          >
            <div
              class="css-o3ijz8"
            >
              <svg
                fill="none"
                height="72"
                viewBox="0 0 61 72"
                width="61"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.4336 17.614C12.4336 7.88457 20.4352 0 30.3091 0C40.1829 0 48.1905 7.88457 48.1905 17.614C48.1905 27.3492 40.1829 35.2338 30.3091 35.2338C20.4352 35.2338 12.4336 27.3492 12.4336 17.614ZM30.4429 72.0003C42.2478 72.0003 52.931 67.3433 60.6312 59.8513V54.5366C60.6312 48.1099 55.2047 42.8941 48.505 42.8941H12.1262C5.42649 42.8941 0 48.1099 0 54.5366V59.5836C7.71835 67.2269 18.5046 72.0003 30.4429 72.0003Z"
                  fill="#005CFF"
                  fill-opacity="0.8"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-1ch8wrx"
          >
            <p
              class="chakra-text css-1j4e3xu"
            >
              Player3
            </p>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-1qn2z3z"
            />
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1qwa78t"
              >
                0
              </p>
              <p
                class="chakra-text css-84i6iv"
              >
                pt
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-7xthq6"
    >
      <p
        class="chakra-text css-1y05cao"
      >
        Out
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`LogPlayerArea スナップショット/自分のターンではない場合 1`] = `
<DocumentFragment>
  <div
    class="css-dvxtzn"
  >
    <div
      class="css-79elbk"
    >
      <div
        class="css-aryzif"
      >
        <div
          class="css-shjrb7"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-04.svg"
          />
        </div>
        <div
          class="css-fni08i"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Clubs-09.svg"
          />
        </div>
        <div
          class="css-1bd1kgk"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-10.svg"
          />
        </div>
        <div
          class="css-2ikapx"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Spades-11.svg"
          />
        </div>
        <div
          class="css-16c4tub"
        >
          <img
            alt="card"
            class="chakra-image css-0"
            src="/card/Hearts-12.svg"
          />
        </div>
      </div>
      <div
        class="css-8myd60"
      >
        <div
          class="css-19tvp9s"
        >
          <p
            class="chakra-text css-1876eo0"
          >
            High Card
          </p>
          <img
            alt="bubble"
            class="chakra-image css-13pr30p"
            src="/bubble.svg"
          />
        </div>
        <div
          class="css-70qvj9"
        >
          <div
            class="css-e18jq9"
          >
            <div
              class="css-o3ijz8"
            >
              <svg
                fill="none"
                height="72"
                viewBox="0 0 61 72"
                width="61"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.4336 17.614C12.4336 7.88457 20.4352 0 30.3091 0C40.1829 0 48.1905 7.88457 48.1905 17.614C48.1905 27.3492 40.1829 35.2338 30.3091 35.2338C20.4352 35.2338 12.4336 27.3492 12.4336 17.614ZM30.4429 72.0003C42.2478 72.0003 52.931 67.3433 60.6312 59.8513V54.5366C60.6312 48.1099 55.2047 42.8941 48.505 42.8941H12.1262C5.42649 42.8941 0 48.1099 0 54.5366V59.5836C7.71835 67.2269 18.5046 72.0003 30.4429 72.0003Z"
                  fill="#005CFF"
                  fill-opacity="0.8"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
          <div
            class="css-1ch8wrx"
          >
            <p
              class="chakra-text css-1j4e3xu"
            >
              Player2
            </p>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-1qn2z3z"
            />
            <div
              class="css-gmuwbf"
            >
              <p
                class="chakra-text css-1qwa78t"
              >
                19,358
              </p>
              <p
                class="chakra-text css-84i6iv"
              >
                pt
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-7xthq6"
    />
    <div
      class="css-z4tay6"
    >
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          fee
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          200
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          1st bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          442
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        +
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          2nd bet
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          0
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        =
      </p>
      <div
        class="css-dvxtzn"
      >
        <p
          class="chakra-text css-c2y0x0"
        >
          total
        </p>
        <p
          class="chakra-text css-tb8v7v"
        >
          642
        </p>
      </div>
      <p
        class="chakra-text css-tb8v7v"
      >
        pt
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
